### DISCUSSION TITLE
Project structure for feature-like deployments

### DISCUSSION THREAD

**Discussion by joaostorrer** [theory/sqitch#783] [General]
Hi, 
I have a project that is deployed to multiple databases of the same engine, each database represents one customer.
This project consists of common objects and objects related to some feature.
Each customer can request different features, so the deployment is different for each one.
My question is what is the best approach to deal with this.
I started creating one sqitch project per feature, but the deployment was getting kind of messy (i had to call sqitch deploy for every sqitch project)
Then I tought of put all in one sqitch project and separate the deployment creating one plan file per customer (but this makes difficult to add/rework database objects).
I tought of separate one plan file per feature as well, but it gets the same situation as multiple sqitch projects.
I've read [sqitch-configuration](https://sqitch.org/docs/manual/sqitch-configuration/) many times, but I'm still a little confused.

**Comment by theory** [comment]
Can you use Variables to conditionally apply changes? I don't know which engine you're using, but the Postgres client, for example, supports [conditional expressions](https://www.postgresql.org/docs/current/app-psql.html#PSQL-METACOMMAND-IF), so you could do something like this in your deploy scripts:

``` sql
\if :feature_a
-- Add feature a
\endif
```

You'd then set the variable using `--set`:

``` sh
sqitch deploy --set feature_a=1 db:pg:cust1
```

Or add a customer-specific target to your configuration:

``` sh
sqitch target add cust1 db:pg:cust1 --set feature_a=1
```

Then it's always set for when you deploy to that target:

``` sh
sqitch deploy cust1
```

**Comment by joaostorrer** [comment]
Using variables the deploy just got better now.
Thanks!
